sources/engine/Stride.Rendering/Rendering/Images/ImageEffect.cs

```diff
@@ -143,12 +148,79 @@ public void SetDepthOutput(Texture depthStencilView, params Texture[] renderTarg
         }
 
         /// <summary>
-        /// Sets the viewport to use .
+        /// Sets the viewport to use.
+        /// </summary>
+        /// <value>The viewport.</value>
+        public void SetViewport(Viewport? value)
+        {
+            if (!value.HasValue)
+            {
+                boundViewportCount = 0;
+            }
+            else
+            {
+                boundViewportCount = 1;
+                viewports[0] = value.Value;
+            }
+        }
+
+        /// <summary>
+        /// Sets the viewport to use.
         /// </summary>
-        /// <param name="viewport">The viewport.</param>
-        public void SetViewport(Viewport? viewport)
+        /// <param name="values">The viewports.</param>
+        public void SetViewports(Viewport[] values)
         {
-            this.viewport = viewport; // TODO: support multiple viewport?
+            SetViewports(values.Length, values);
+        }
+
+        /// <summary>
+        /// Sets the viewport to use.
+        /// </summary>
+        /// <param name="viewportCount">The viewports.</param>
+        /// <param name="values">The viewports.</param>
+        public void SetViewports(int viewportCount, Viewport[] values)
+        {
+            boundViewportCount = viewportCount;
+            if(viewportCount > 0)
+                Array.Copy(values, 0, this.viewports, 0, boundViewportCount);
+        }
+
+        /// <summary>
+        /// Binds a single scissor rectangle to the rasterizer stage. See <see cref="Render+states"/> to learn how to use it.
```

Comment:
Is this a valid `cref` reference?

Reply:
Nope these comments are copied from CommandList. Basically most of this logic was copied from there :)

Reply:
Ok, let's leave it for the moment